### PR TITLE
GIT-3004: Merge request create -> transition an issue script was impl…

### PR DESCRIPTION
**Pull request by adokaeva**
Script was implemented in suggestion that JIRA Issue Key is present in merge request title. 
First issue's key from merge/pull request title will be taken as issue for transition. 
Issue's status will be changed on behalf of issue's assignee.

### CODE REVIEW

File: scripts/utils.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    var status = issue.getStatus();
+    var workFlow = workflowManager.getWorkflow(issue);
+    var currentStep = workFlow.getLinkedStep(status);
+    return currentStep.getActions();
+}
+
+function getIdForStatusWithName(statusName, possibleActionsList) {
+    var actionId = 0;
+    for each (var actionDescriptor in possibleActionsList) {
+       	if(actionDescriptor.getName() === statusName) {
```
Comment (dmalyshkin-issart) [inline scripts/utils.js]: Indents are different. Please check your IDE settings to use only spaces as indent characters.
If you prefer to use tabs -- please use *only* tabs everywhere (and therefore please convert existing space indents to tabs too).
Please do not mix tabs and spaces in indents.

File: scripts/on-merge-req-created.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+//# sourceURL=on-merge-req-created.js
+'use strict';
+
```
Comment (dmalyshkin-issart) [inline scripts/on-merge-req-created.js:3]: Please add a description for this script:
 - The goal of the script.
 - What does this script do.
 - Which parameters should be configured by Jira admin before script using.

File: scripts/on-merge-req-created.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+var workflowManager = getComponent("com.atlassian.jira.workflow.WorkflowManager");
+var issueManager = getComponent("com.atlassian.jira.issue.IssueManager");
+
+var pullreq = JSON.parse(pullreqJson);
+var issueKey = getIssueKey(pullreq.title);
+if(issueKey !== null) {
+    var issue = issueManager.getIssueByCurrentKey(issueKey);
+    var status = issue.getStatus();
+    var user = issue.getAssignee();
+    print(status);
```
Comment (dmalyshkin-issart) [inline scripts/on-merge-req-created.js]: Do you need this debug output on production?

File: scripts/on-merge-req-created.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        var newStatusId = getIdForStatusWithName(IN_PROGRESS, possibleActionsList);
+        if(newStatusId) {
+            var issueService = getComponent("com.atlassian.jira.bc.issue.IssueService");
+            var transitionValidationResult = issueService.validateTransition(
+            user, issue.getId(), newStatusId, issueService.newIssueInputParameters()
+            );
+            if (transitionValidationResult.isValid()) {
+                var transitionResult =  issueService.transition(user, transitionValidationResult);
+            } else {
+                print(transitionValidationResult.getErrorCollection());
```
Comment (dmalyshkin-issart) [inline scripts/on-merge-req-created.js]: You print an error without any additional comments.
I think a note about the action which were performing will be useful.

File: scripts/utils.js (hunk 1 above)
Comment (adokaeva) [reply scripts/utils.js]: done

File: scripts/on-merge-req-created.js (hunk 2 above)
Comment (adokaeva) [reply scripts/on-merge-req-created.js:3]: done

File: scripts/on-merge-req-created.js (hunk 4 above)
Comment (adokaeva) [reply scripts/on-merge-req-created.js]: done

File: scripts/on-merge-req-created.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
+//# sourceURL=on-merge-req-created.js
+'use strict';
+
+//  Description
+//
+// This script work as a handler that checks if an issue is in open status.
```
Comment (dmalyshkin-issart) [inline scripts/on-merge-req-created.js]: I've slightly re-phrased the description:

//  This script does an automatic issue transition to "In progress" status if
// the issue is still in "Open" status when a merge/pull request has been created for the issue.
// It works on the assumption that an issue key is present in the title of the merge/pull request.
// First issue's key from the merge/pull request title will be taken as an issue for the transition.
// Issue's status will be changed on behalf of the issue's assignee.

File: scripts/utils.js (hunk 1 above)
Comment (dmalyshkin-issart) [reply scripts/utils.js]: I don't see the change for utils.js -- http://take.ms/3Vxqb

File: scripts/on-merge-req-created.js (hunk 5 above)
Comment (adokaeva) [reply scripts/on-merge-req-created.js]: done

File: scripts/utils.js (hunk 1 above)
Comment (adokaeva) [reply scripts/utils.js]: done

**Review by dmalyshkin-issart — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**